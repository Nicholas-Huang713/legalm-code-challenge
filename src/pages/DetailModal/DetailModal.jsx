import React from 'react'
import Modal from '../../components/Modal/Modal';
import {useSelector} from 'react-redux';
import './DetailModal.scss';

export default function DetailModal({isOpen, handleCloseDetailModal}) {
    const { dog, owners } = useSelector((state) => state.dog.dogs);

  return (
    <Modal isOpen={isOpen} onClose={handleCloseDetailModal}>
        {dog.name && owners.length > 0 ? 
        <div className='modal-content-wrapper'>
            <header>
                <h2>{dog.name}</h2>
            </header>
            <div className='modal-section'>
                <div className='modal-spacing'><strong>Owner:</strong> {owners[0].name}</div>
                <div><strong>Exp:</strong> {owners[0].exp}yrs</div>
            </div>
            <div>
                <img src={dog.img} className='modal-image'/>
            </div>
            <div>
            <strong>Favorite Food:</strong> {dog.food}
            </div>
        </div>
            : null
        }

    </Modal>
  )
}
