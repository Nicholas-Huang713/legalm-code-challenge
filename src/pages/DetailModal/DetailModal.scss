.modal-content-wrapper {
    overflow-y: auto;
}

.modal-section {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
    margin-bottom: 1em;
}

.modal-spacing {
    margin-right: 2em;
}

.modal-image {
    max-width: 50%;
}