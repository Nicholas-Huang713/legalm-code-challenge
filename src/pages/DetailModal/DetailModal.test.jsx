import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import DetailModal from "./DetailModal";
const thunkMiddleware = require("redux-thunk").thunk;
const mockStore = configureMockStore([thunkMiddleware]);
const mockCloseDetailModal = vi.fn();
import {
  emptyDogDataInStore,
  dogDataInStore,
  ownersList,
} from "../../test/mocks/mockValues";

let store;

const initStoreNoDogData = () => {
  store = mockStore({
    owner: ownersList,
    dog: emptyDogDataInStore,
  });
  store.dispatch = vi.fn();
};

const initStoreWithDogData = () => {
  store = mockStore({
    owner: ownersList,
    dog: dogDataInStore,
  });
  store.dispatch = vi.fn();
};

describe("DetailModal component", () => {
  it("renders no data text when no dog data to display", () => {
    initStoreNoDogData();
    render(
      <Provider store={store}>
        <DetailModal
          isOpen={true}
          handleCloseDetailModal={mockCloseDetailModal}
        />
      </Provider>
    );
    expect(screen.getByText("No data to display")).toBeInTheDocument();
  });

  it("renders modal with dog data", () => {
    initStoreWithDogData();
    const { container } = render(
      <Provider store={store}>
        <DetailModal
          isOpen={true}
          handleCloseDetailModal={mockCloseDetailModal}
        />
      </Provider>
    );
    expect(screen.getByText("peanut")).toBeInTheDocument();
    expect(screen.getByText("pizza")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
