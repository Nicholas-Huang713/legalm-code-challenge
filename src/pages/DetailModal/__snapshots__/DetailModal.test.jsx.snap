// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DetailModal component > renders modal with dog data 1`] = `
<div>
  <div
    class="modal-backdrop"
    data-testid="modal-backdrop"
  >
    <div
      class="modal-content"
    >
      <button
        class="close-btn"
      >
        X
      </button>
      <section
        class="modal-content-main"
      >
        <div
          class="modal-content-wrapper"
        >
          <header>
            <h2>
              peanut
            </h2>
          </header>
          <div
            class="modal-section"
          >
            <div
              class="modal-spacing"
            >
              <strong>
                Owner:
              </strong>
               
              John Doe
            </div>
            <div>
              <strong>
                Exp:
              </strong>
               
              3
              yrs
            </div>
          </div>
          <div>
            <img
              class="modal-image"
            />
          </div>
          <div>
            <strong>
              Favorite Food:
            </strong>
             
            pizza
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
